synthesizer/process/src/stack/evaluate.rs

```diff
@@ -97,15 +97,20 @@ impl<N: Network> Stack<N> {
     ///
     /// # Errors
     /// This method will halt if the given inputs are not the same length as the input statements.
-    pub fn evaluate_function<A: circuit::Aleo<Network = N>>(
+    pub fn evaluate_function<A: circuit::Aleo<Network = N>, R: CryptoRng + Rng>(
         &self,
-        call_stack: CallStack<N>,
+        mut call_stack: CallStack<N>,
         caller: Option<ProgramID<N>>,
+        root_tvk: Option<Field<N>>,
+        rng: &mut R,
     ) -> Result<Response<N>> {
         let timer = timer!("Stack::evaluate_function");
 
         // Retrieve the next request, based on the call stack mode.
-        let (request, call_stack) = match &call_stack {
+        let (request, call_stack) = match &mut call_stack {
+            CallStack::Authorize(ref mut requests, ..) => {
+                (requests.pop().ok_or_else(|| anyhow!("No more requests in the call stack."))?, call_stack)
```

Comment:
Why not use `call_stack.pop()?;` like in `execute_function` to get the request?

Reply:
Ah much cleaner! 7258279cd